BankChain/app/src/main/java/nl/tudelft/ewi/ds/bankchain/activities/RecentTransactionsActivity.java

```diff
@@ -114,15 +129,17 @@ public void showRecentTransactions(List<? extends Transaction> transactionList)
             if (transactionList.get(i).getValue() != null && transactionList.get(i).getCurrency() != null) {
                 details.add(transactionList.get(i).getValue().toString() + " " + transactionList.get(i).getCurrency().toString());
             }
-            if (transactionList.get(i).getDirection() != null) {
-                details.add("" + transactionList.get(i).getDirection().toString());
+            if (transactionList.get(i).getValue() < 0) {
+                details.add("OUT");
+            } else {
+                details.add("IN");
             }
 //                     TODO: implement once getDate functions correctly
 //                    if (transactionList.get(i).getDate() != null) {
 //                        details.add(transactionList.get(i).getDate().toString());
 //                    }
-            if (transactionList.get(i).getCounterparty() != null) {
-                details.add("" + transactionList.get(i).getCounterparty().toString());
+            if (transactionList.get(i).getCounterAccount() != null) {
+                details.add("" + transactionList.get(i).getCounterAccount().getParty().toString());
```

Comment:
Is it possible for `getCounterAccount()` to be not `null`, but `getCounterAccount().getParty()` to be null?